sql/parse/create_index.go

```diff
@@ -63,10 +73,128 @@ func parseCreateIndex(s string) (sql.Node, error) {
 		name,
 		plan.NewUnresolvedTable(table),
 		indexExprs,
-		"",
+		driver,
+		config,
 	), nil
 }
 
+func optional(steps ...parseFunc) parseFunc {
+	return func(rd *bufio.Reader) error {
+		for _, step := range steps {
+			err := step(rd)
+			if err == io.EOF || errUnexpectedSyntax.Is(err) {
+				return nil
+			}
+
+			if err != nil {
+				return err
+			}
+		}
+
+		return nil
+	}
+}
+
+func readKeyValue(kv map[string]string) parseFunc {
+	return func(rd *bufio.Reader) error {
+		r, _, err := rd.ReadRune()
+		if err != nil {
+			return err
+		}
+
+		if r != '(' {
+			return errUnexpectedSyntax.New("(", string(r))
+		}
+
+		for {
+			var key, value string
+			steps := []parseFunc{
+				skipSpaces,
+				readIdent(&key),
+				skipSpaces,
+				expectRune('='),
+				skipSpaces,
+				readValue(&value),
+				skipSpaces,
+			}
+
+			for _, step := range steps {
+				if err := step(rd); err != nil {
+					return err
+				}
+			}
+
+			r, _, err := rd.ReadRune()
+			if err != nil {
+				return err
+			}
+
+			switch r {
+			case ')':
+				kv[key] = value
+				return nil
+			case ',':
+				kv[key] = value
+				continue
+			default:
+				return errUnexpectedSyntax.New(", or )", string(r))
+			}
+		}
+	}
+}
+
+func readValue(val *string) parseFunc {
+	return func(rd *bufio.Reader) error {
+		var buf bytes.Buffer
+		var singleQuote, doubleQuote, ignoreNext bool
+		var first = true
+		for {
+			r, _, err := rd.ReadRune()
+			if err == io.EOF {
+				break
+			}
+
+			if err != nil {
+				return err
+			}
+
+			if singleQuote || doubleQuote {
+				switch true {
```

Comment:
In this case, what `ignoreNext` does is ignore the meaning of the next character, but we need everything written, that's why there is no `break` in there. We need every character written because we delegate the real parsing to `sqlparser`.